### PR TITLE
Add article introduction aux hooks

### PR DESCRIPTION
**Pull request by MehdiDr**

### CODE REVIEW

**Comment by ElevenWilson** [comment]
17 words might be misspelled, please check them: cliquer, componentDidMount, componentDidUpdate, componentWillUnmount, HOC, lang, lifecycle, LOGIN, nommage, PureComponent, redux, renderProps, useContext, UseEffect, useMousePosition, useReducer, useState

**Comment by ElevenWilson** [comment]
17 words might be misspelled, please check them: cliquer, componentDidMount, componentDidUpdate, componentWillUnmount, HOC, lang, lifecycle, mehdidr, nommage, PureComponent, redux, renderProps, useContext, UseEffect, useMousePosition, useReducer, useState

**Comment by ElevenWilson** [comment]
17 words might be misspelled, please check them: cliquer, componentDidMount, componentDidUpdate, componentWillUnmount, HOC, lang, lifecycle, mehdidr, nommage, PureComponent, redux, renderProps, useContext, UseEffect, useMousePosition, useReducer, useState

**Comment by ElevenWilson** [comment]
17 words might be misspelled, please check them: cliquer, componentDidMount, componentDidUpdate, componentWillUnmount, HOC, lang, lifecycle, mehdidr, nommage, PureComponent, redux, renderProps, useContext, UseEffect, useMousePosition, useReducer, useState

**Comment by ElevenWilson** [comment]
17 words might be misspelled, please check them: cliquer, componentDidMount, componentDidUpdate, componentWillUnmount, HOC, lang, lifecycle, mehdidr, nommage, PureComponent, redux, renderProps, useContext, UseEffect, useMousePosition, useReducer, useState

**Comment by ElevenWilson** [comment]
17 words might be misspelled, please check them: cliquer, componentDidMount, componentDidUpdate, componentWillUnmount, HOC, lang, lifecycle, mehdidr, nommage, PureComponent, redux, renderProps, useContext, UseEffect, useMousePosition, useReducer, useState

**Comment by ElevenWilson** [comment]
17 words might be misspelled, please check them: cliquer, componentDidMount, componentDidUpdate, componentWillUnmount, HOC, lang, lifecycle, mehdidr, nommage, PureComponent, redux, renderProps, useContext, UseEffect, useMousePosition, useReducer, useState

File: _posts/fr/2019-04-02-introduction-aux-hooks.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+```
+
+Ce code peut vite poser problème, principalement pour deux raisons :
+
+- Si le mouvement de la souris est nécessaire dans un autre composant, il va falloir dupliquer ce code.
+- Si on complexifie le composant en ajoutant d’autres comportements de la souris, cela peut rapidement devenir illisible, notamment à cause du contenu du `componentDidMount` et du `componentWillUnmount`.
+
+Voilà comment on peut régler le problème avec les deux méthodes dont nous avons parlé plus haut :
+
+### HOC
```
Comment (elfakamal) [inline _posts/fr/2019-04-02-introduction-aux-hooks.md]:
```suggestion
### HOC (Higher Order Component)
```

**Comment by ElevenWilson** [comment]
17 words might be misspelled, please check them: cliquer, componentDidMount, componentDidUpdate, componentWillUnmount, HOC, lang, lifecycle, mehdidr, nommage, PureComponent, redux, renderProps, useContext, UseEffect, useMousePosition, useReducer, useState

File: _posts/fr/2019-04-02-introduction-aux-hooks.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  )
+}
+
+export default MouseRender;
+```
+
+Comme on peut le voir, on utilise 2 hooks dans ce code :
+
+- `useState`, comme son nom l’indique, permet d’utiliser le state : le premier argument étant le state initial (qui n’a pas nécessairement besoin d’être un objet), et le second étant la fonction à appliquer sur cette valeur.
+- `UseEffect` indique que le composant peut causer des effets de bord comme par exemple fetcher des données ou modifier manuellement le DOM et vient remplacer `componentDidMount`, `componentDidUpdate` et `componentWillUnmount` dans la même API.
```
Comment (elfakamal) [inline _posts/fr/2019-04-02-introduction-aux-hooks.md]:
```suggestion
- `useEffect` indique que le composant peut causer des effets de bord comme par exemple fetcher des données ou modifier manuellement le DOM et vient remplacer `componentDidMount`, `componentDidUpdate` et `componentWillUnmount` dans la même API.
```

**Comment by ElevenWilson** [comment]
18 words might be misspelled, please check them: cliquer, componentDidMount, componentDidUpdate, componentWillUnmount, HOC, jpg, lang, lifecycle, mehdidr, nommage, PureComponent, redux, renderProps, useContext, useEffect, useMousePosition, useReducer, useState

**Comment by ElevenWilson** [comment]
18 words might be misspelled, please check them: cliquer, componentDidMount, componentDidUpdate, componentWillUnmount, HOC, jpg, lang, lifecycle, mehdidr, nommage, PureComponent, redux, renderProps, useContext, useEffect, useMousePosition, useReducer, useState

**Pull request closed without merging**